src/serac/physics/solid_mechanics.hpp

```diff
@@ -363,6 +372,137 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
     bcs_.addEssential(disp_bdr, component_disp_bdr_coef_, displacement_.space(), component);
   }
 
+  /**
+   * @brief Set the displacement essential boundary conditions on a set of true degrees of freedom
+   *
+   * @param true_dofs A set of true degrees of freedom to set the displacement on
+   * @param disp The vector function containing the prescribed displacement values
+   *
+   * The @a true_dofs list can be determined using functions from the @a mfem::ParFiniteElementSpace class.
+   *
+   * @note The coefficient is required to be vector-valued. However, only the dofs specified in the @a true_dofs
+   * array will be set. This means that if the @a true_dofs array only contains dofs for a specific vector component in
+   * a vector-valued finite element space, only that component will be set.
```

Comment:
I do not have any case in which I want to constrain one or two components only, but good to know that I have control over that with this method for less common configurations. I guess that defining the coefficient as be vector-valued may confuse a new user (that doesn't read the documentation)... Just trying to add a meaningful comment here, but not a big deal for me. Happy to be able to use it now :). Thanks!